java/dagger/internal/codegen/SimpleMethodBindingExpression.java

```diff
@@ -176,13 +176,16 @@ private Expression injectMembers(CodeBlock instance) {
   }
 
   private Optional<CodeBlock> moduleReference(ClassName requestingClass) {
-    return provisionBinding.requiresModuleInstance()
-        ? provisionBinding
-            .contributingModule()
-            .map(Element::asType)
-            .map(ComponentRequirement::forModule)
-            .map(module -> componentRequirementExpressions.getExpression(module, requestingClass))
-        : Optional.empty();
+    if (provisionBinding.requiresModuleInstance(compilerOptions)) {
+      return provisionBinding.contributingModule()
+          .map(Element::asType)
+          .map(ComponentRequirement::forModule)
+          .map(module -> componentRequirementExpressions.getExpression(module, requestingClass));
+    } else if (compilerOptions.kotlinMetadata() && provisionBinding.isModuleKotlinObject()) {
+      return Optional.of(CodeBlock.of("$T.INSTANCE", provisionBinding.bindingTypeElement().get()));
+    } {
```

Comment:
:)

Can you clean this up?

Reply:
clean up style or something in the logic? On mobile so forgive me if I'm missing something obvious :)

Reply:
You've got a `} {`, which is probably intended to be `} else {` or just not nested in braces at all